go-controller/pkg/ovn/master.go

```diff
@@ -606,12 +606,12 @@ func (oc *DefaultNetworkController) syncNodes(kNodes []interface{}) error {
 
 	staleNodes := sets.NewString()
 	for _, nodeSwitch := range nodeSwitches {
-		if !foundNodes.Has(nodeSwitch.Name) {
+		if nodeSwitch.Name != types.TransitSwitch && !foundNodes.Has(nodeSwitch.Name) {
```

Comment:
Great find!
NIT: Could we rename `nodeSwitches`/`nodeSwitch` to something else?
It is a little confusing and I think it might be the reason we missed the bug in the first place.

Reply:
can we merge as is to unblock CI and then i'll submit another PR for the cosmetic change?